Modules/_testinternalcapi.c

```diff
@@ -35,6 +36,38 @@ get_recursion_depth(PyObject *self, PyObject *Py_UNUSED(args))
 }
 
 
+static PyObject*
+codecs_unregister(PyObject *self, PyObject *args)
+{
+    PyObject *search_function;
+
+    if (!PyArg_ParseTuple(args, "O:codecs_unregister",
+                          &search_function)) {
+        return NULL;
+    }
+    if (search_function == NULL) {
+        return NULL;
+    }
+
+    PyInterpreterState *interp = PyInterpreterState_Get();
+    if (interp->codec_search_path == NULL) {
+        return NULL;
```

Comment:
This function cannot be called before _PyUnicode_InitEncodings() which initiallizes codec_search_path.

**Maybe** the function could be called during PyInterpreterState_Clear() which calls Py_CLEAR(interp->codec_search_path), but if codec_search_path is NULL here, I suggest to simply do nothing: return None.

Reply:
Ok, Make sense.